packages/shared-base-ui/src/hooks/useValueRef.ts

```diff
@@ -12,6 +13,24 @@ export function useValueRef<T>(ref: ValueRef<T>): Readonly<T> {
     )
 }
 
+export function useValueRefReactQuery<T>(key: `@@${string}`, ref: ValueRefWithReady<T>) {
+    const { data, refetch } = useQuery({
+        queryKey: [key],
+        queryFn: async () => {
+            await ref.readyPromise
+            return ref.value
+        },
+        placeholderData: () => ref.value as any,
+        networkMode: 'always',
+    })
+    useEffect(() => {
+        refetch()
+        ref.addListener(() => refetch())
```

Comment:
Do you forget to return the clean-up callback?

Reply:
nice catch thanks!